templates/SilverStripe/Admin/Includes/BrowserWarning.ss

```diff
@@ -0,0 +1,31 @@
+<!--[if lte IE 9]>
+    <link rel="stylesheet" href="{$BaseURL}silverstripe-admin/client/dist/styles/browser-warning.css">
+<![endif]-->
+
+<div id="browser-warning-wrapper" class="browser-warning-wrapper hide">
+    <div class="browser-warning vertical-align-items">
+        <div class="browser-warning__content flexbox-area-grow">
+            <p class="ss-logo">SilverStripe</p>
+            <h1><%t SilverStripe\Admin\LeftAndMain.BROWSER_WARNING_HEADING 'Unsupported browser' %></h1>
+
+            <p>
+                <%t SilverStripe\Admin\LeftAndMain.BROWSER_WARNING_TEXT 'You&apos;re using a web browser that&apos;s not supported by SilverStripe.<br />Please upgrade to one of the following options.' %>
+            </p>
+            <ul class="browser-warning__list">
+                <li data-browser="ie"><a href="https://support.microsoft.com/en-us/help/17621/internet-explorer-downloads">Internet Explorer(11+)</a></li>
+
+                <li data-browser="firefox"><a href="https://www.mozilla.org/en-US/firefox/">Mozilla Firefox</a></li>
+
+                <li data-browser="chrome"><a href="https://www.google.com/chrome/">Google Chrome</a></li>
+            </ul>
+        </div>
+    </div>
+</div>
+
+<% require javascript("silverstripe-admin/client/dist/js/browserWarning.js") %>
+
+<!--[if lte IE 8]>
+    <script>
+        document.getElementById('browser-warning-wrapper').className = 'browser-warning-wrapper';
```

Comment:
is this needed because browser-detect doesn't detect ie8 properly? almost seems counter-intuitive for this to happen...

Reply:
This is because the script in `browserWarning.js` doesn't execute on IE8 (to be honest I'm not sure why exactly).